packages/shared/src/hooks/useTimedAnimation.ts

```diff
@@ -0,0 +1,87 @@
+import { useCallback, useEffect, useRef, useState } from 'react';
+import { isNullOrUndefined } from '../lib/func';
+
+interface UseTimedAnimation {
+  timer: number;
+  isAnimating: boolean;
+  endAnimation: () => void;
+}
+
+interface UseTimedAnimationProps {
+  autoEndAnimation: boolean;
+  animationDuration: number;
+  onAnimationEnd?: () => void;
+  onAnimationStop?: () => void;
+}
+
+const TEMPORARY_ID = 1;
+const INTERVAL_COUNT = 10;
+const IN_OUT_ANIMATION = 140;
+
+export const useTimedAnimation = ({
+  autoEndAnimation = true,
+  animationDuration,
+  onAnimationEnd,
+  onAnimationStop,
+}: UseTimedAnimationProps): UseTimedAnimation => {
+  const timeout = useRef<number>();
+  const [timer, setTimer] = useState(0);
+  const [intervalId, setIntervalId] = useState<number>(null);
+
+  const endAnimation = useCallback(() => {
+    if (!isNullOrUndefined(animationDuration)) {
+      return;
+    }
+
+    window.clearInterval(intervalId);
```

Comment:
We don't set the `intervalId` to null here. I suggest a function that both clears the interval and sets the variable to null

Reply:
Yes, that was intentional so we can allow the out animation to process first before dismounting. It is then set to null in the line below:
https://github.com/dailydotdev/apps/blob/7efbf1e531f4484524f980d3730d8a42c733a900/packages/shared/src/hooks/useTimedAnimation.ts#L45